tf_encrypted/protocol/aby3/model/logistic_regression.py

```diff
@@ -0,0 +1,297 @@
+import tensorflow as tf
+import tf_encrypted as tfe
+
+
+class LogisticRegression:
+    def __init__(self, num_features, init_learning_rate=0.01):
+        self.w = tfe.define_private_variable(
+            tf.random_uniform([num_features, 1], -0.01, 0.01))
+        self.b = tfe.define_private_variable(tf.zeros([1]))
+        self.init_learning_rate = init_learning_rate
+
+    @property
+    def weights(self):
+        return self.w, self.b
+
+    def forward(self, x):
+        with tf.name_scope("forward"):
+            out = tfe.matmul(x, self.w) + self.b
+            y_hat = tfe.sigmoid(out)
+            return y_hat
+
+    def backward(self, x, dy, learning_rate):
+        batch_size = x.shape.as_list()[0]
+        with tf.name_scope("backward"):
+            dw = tfe.matmul(tfe.transpose(x), dy) / batch_size
+            db = tfe.reduce_sum(dy, axis=0) / batch_size
+            assign_ops = [
+                tfe.assign(self.w, self.w - dw * learning_rate),
+                tfe.assign(self.b, self.b - db * learning_rate)
+            ]
+            return assign_ops
+
+    def loss_grad(self, y, y_hat):
+        with tf.name_scope("loss-grad"):
+            dy = y_hat - y
+            return dy
+
+    def fit_batch(self, x, y):
+        with tf.name_scope("fit-batch"):
+            y_hat = self.forward(x)
+            dy = self.loss_grad(y, y_hat)
+            fit_batch_op = self.backward(x, dy, self.init_learning_rate)
+            return fit_batch_op
+
+    def fit(self, sess, x, y, num_batches):
+        fit_batch_op = self.fit_batch(x, y)
+        for batch in range(num_batches):
+            sess.run(fit_batch_op, tag="fit-batch")
+
+    def loss(self, sess, x, y, player_name):
+        def print_loss(y_hat, y):
```

Comment:
do these outputs in cleartext break privacy? (cc @mortendahl)

Reply:
> do these outputs in cleartext break privacy? (cc @mortendahl)

Yes, I leave it here only for debugging. Normal runs should not reveal loss.

Reply:
a comment to that effect would be much appreciated :)

Reply:
Tracked via https://github.com/tf-encrypted/tf-encrypted/issues/737